.github/workflows/flake8.yml

```diff
@@ -0,0 +1,31 @@
+name: flake8 
+
+on:
+  push:
+    paths:
+      - '**.py'
+  pull_request:
+    paths:
+      - '**.py'
+
+jobs:
+  flake8_py3:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.9
+          architecture: x64
+      - name: Install flake8
+        run: |
+          pip install --upgrade pip
+          pip install -r REQUIREMENTS.qa.txt
+      - name: Run flake8
+        uses: suo/flake8-github-action@releases/v1
+        with:
+          checkName: 'flake8_py3'   # NOTE: this needs to be the same as the job name
```

Comment:
I am not a fan of these complex `uses: …` github actions, when the alternative is to just pip install flake and run it.
This also applies to the `black` case, but there it is at least possible to configure the path to check. For this case I did not see a way to do that.

I've added a commit that just installs and runs flake8 manually and then also checks the files in `bin/`.
If you agree with that change it would be great if you squashed it into your commit.

Reply:
On the other hand we have already established workflows in our USB-SD-Mux tool. I would like to stick as close to those as possible: https://github.com/linux-automation/usbsdmux/tree/master/.github/workflows
(Or make the same changes there.)